### PR TITLE
Release post for Jekyll v3.5

### LINKED ISSUE #6074: Release Jekyll v3.5
Hey @jekyll/core & @jekyll/affinity-team-captains!

I just saw the History.markdown file for `HEAD` and MY GOODNESS y'all have been busy. Thank you. 💞 Think it's time for a v3.5.0? Anything in particular we're waiting for before the release?

Let me know. 👂 

❤️ :octocat:

### PR DESCRIPTION
**Pull request by parkr**
It's time to release v3.5.0! Here's a draft for a release post.

Fixes #6074.

/cc @jekyll/documentation

### CODE REVIEW

File: docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+There was a huge amount of effort put into this release by our maintainers,
+especially @pathawks, @DirtyF, and @pup. Huge thanks to them for ushering
+this release along and keeping the contributions flowing! Jekyll wouldn't
+work without the tireless dedication of our team captains & maintainers.
+Thank you, all!
+
+A huge thanks as well to our NNN contributors to this release: Adam Hollett, Aleksander Kuś, Alfred Myers, Anatoliy Yastreb, Antonio Argote, Ashton Hellwig, Ashwin Maroli, Ben Balter, BlueberryFoxtrot, Brent Yi, Chris Finazzo, Christoph Päper, Christopher League, Chun Fei Lung, Colin, David Zhang, Eric Leong, Finn Ellis, Florian Thomas, Frank Taillandier, Hendrik Schneider, Henry Kobin, Ivan Storck, Jakub Klímek, Jan Pobořil, Jeff Puckett, Jonathan Hooper, Kaligule, Kevin Funk, Krzysztof Szafranek, Liu Cheng, Lukasz Brodowski, Marc Bruins, Marcelo Canina, Martin Desrumaux, Mer, Nate, Oreonax, Parker Moore, Pat Hawks, Pedro Lamas, Phil Nash, Ricardo N Feliciano, Ricky Han, Roger Sheen, Ryan Lue, Ryan Streur, Shane Neuville, Sven Meyer, Tom Johnson, William Entriken, Yury V. Zaytsev, Zarino Zappia, ashmaroli, dyang, jekylltools, sean delaney, zenHeart
```
Comment (pathawks) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: Be sure to replace `NNN` before merging :+1:

**Comment by DirtyF** [comment]
Thanks @parkr for summing all the activity and preparing the release.

Does anyone know a of summary of Liquid 4 features? How are Jekyll users supposed to know what the new version brings to Jekyll? /cc @fw42

File: docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+date: 2017-06-15 17:32:32 -0400
+author: parkr
+version: 3.5.0
+categories: [release]
+---
+
+Good news! Nearly 400 commits later, Jekyll 3.5.0 has been released into
+the wild. Some new shiny things you might want to test out:
+
+- Jekyll now uses Liquid 4, the latest!
```
Comment (pup) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: adding to what @DirtyF said, might be nice to link a summary of liquid 4 features here

**Comment by pup** [comment]
hey, also jekyll reached 30k github stars, how fitting!

File: docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown (hunk 2 above)
Comment (benbalter) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: :+1: to explaining why users should care about liquid 4.

Comment (pup) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: https://github.com/Shopify/liquid/blob/master/History.md#400--2016-12-14--branch-4-0-stable it seems the main feature is dropped support for ruby 2.0?

Comment (adamvoss) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: @pup 😕 Liquid's history does not even mention [whitespace control](https://help.shopify.com/themes/liquid/basics/whitespace) which is what I am most looking forward to from it.  To be sure I did check the issue tracker that it actually exists in v4 (Shopify/liquid#807).

Comment (DirtyF) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: I hope users will benefit from https://github.com/Shopify/liquid/pull/681

File: docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown (hunk 1 above)
Comment (ashmaroli) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: FYI, `Ashwin Maroli` and `ashmaroli` are the same person.. like Jekyll and Hyde.. :trollface:

File: docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- Themes can specify runtime dependencies (in their gemspecs) and we'll require those. This makes it easier for theme writers to use plugins.
+- Speaking of themes, we'll properly handle the discrepancy between a convertible file in the local site and a static file in the theme. Overriding a file locally now doesn't matter if it's convertible or static.
+- Pages, posts, and other documents can now access layout variables via `{% raw %}{{ layout }}{% endraw %}`.
+- The `gems` key in the `_config.yml` is now `plugins`. This is backwards-compatible, as Jekyll will gracefully upgrade `gems` to `plugins` if you use the former.
+- Filters like `sort` now allow you to sort based on a subvalue, e.g. `{% raw %}{% assign sorted = site.posts | sort: "image.alt_text" %}{% endraw %}`.
+- You can now create tab-separated data files.
+- Using `layout: none` will now produce a file with no layout. Equivalent to `layout: nil`, with the exception that `none` is a truthy value and won't be overwritten by front matter defaults.
```
Comment (ashmaroli) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: Error here: `layout: none` is equivalent to `layout: null`

File: docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown (hunk 2 above)
Comment (ashmaroli) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: @adamvoss A commit regarding [`whitespace control`](https://github.com/Shopify/liquid/commit/e1d40c7d89accd91f0d3f9bba0637ee33f708646).
The meta info above the author confirms its existence in `4.0-stable`

Comment (ashmaroli) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: The entire [commit history](https://github.com/Shopify/liquid/compare/3-0-stable...4-0-stable) between 3.0 and 4.0

Comment (pup) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: seems like we're going to have to write our own little changelog...

Comment (DirtyF) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: amended the post with a few highlights and a link to Liquid 4 CHANGELOG

File: docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown (hunk 3 above)
Comment (ashmaroli) [inline docs/_posts/2017-06-14-jekyll-3-5-0-released.markdown]: This edit needs to be made as well..

**Pull request merged**